compiler/rustc_middle/src/hir/mod.rs

```diff
@@ -104,11 +104,13 @@ impl<'tcx> TyCtxt<'tcx> {
         self.parent_module_from_def_id(id.owner.def_id)
     }
 
-    pub fn impl_subject(self, def_id: DefId) -> ImplSubject<'tcx> {
-        self.impl_trait_ref(def_id)
-            .map(|t| t.subst_identity())
-            .map(ImplSubject::Trait)
-            .unwrap_or_else(|| ImplSubject::Inherent(self.type_of(def_id).subst_identity()))
+    pub fn impl_subject(self, def_id: DefId) -> EarlyBinder<ImplSubject<'tcx>> {
+        EarlyBinder(
+            self.impl_trait_ref(def_id)
```

Comment:
nit: maybe we should just `match` here instead of map + map + unwrap... I don't really like subst_identity then re-wrapping in an early binder...

Reply:
Thanks! Yeah that's pretty confusing as is. I changed it to use a `match` and then `map` in the newest commit, please let me know if it looks alright :+1: